plugin/externalPluginEntry.go

```diff
@@ -470,6 +470,8 @@ func (e *externalPluginEntry) Exec(ctx context.Context, cmd string, args []strin
 	cmdObj.SetStderr(execCmd.Stderr())
 	if opts.Stdin != nil {
 		cmdObj.SetStdin(opts.Stdin)
+	} else {
+		cmdObj.SetStdin(strings.NewReader(""))
```

Comment:
I don't like this because it doesn't account for `echo -n | cat` and `cat`'s distinct behavior.

Reply:
When you execute just `cat`, it's waiting for you to supply stdin. This seems like it does capture that difference.

Reply:
Think this should be removed since we've resolved the Stdin issue earlier

Reply:
I think this was still necessary. If there's no Stdin provided, go exec reads `/dev/null` but I think we really want it to say that it's empty and closed.

Reply:
I can back it out if you want to explore this more separately.

Reply:
Ah, that's ok then. A comment explaining what you just said would help.